src/test/java/org/minions/devfund/angela/battleship/ShipTest.java

```diff
@@ -0,0 +1,101 @@
+package org.minions.devfund.angela.battleship;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test for {@link Ship}.
+ */
+public class ShipTest {
+
+    private Ship ship;
+
+    /**
+     * Verifies that empty sea never sunks.
+     */
+    @Test
+    public void testEmptySeaNeverSunks() {
+        ship = new EmptySea();
+        ship.setBowColumn(0);
+        ship.setBowRow(0);
+        assertFalse("There is a ship in that position", ship.shootAt(0, 0));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "There is a ship in that position" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)